android/src/main/java/com/reactnativecommunity/viewpager/VerticalViewPager.java

```diff
@@ -0,0 +1,109 @@
+package com.reactnativecommunity.viewpager;
+
+import android.content.Context;
+import android.util.AttributeSet;
+import android.view.GestureDetector;
+import android.view.MotionEvent;
+import android.view.View;
+
+import android.support.v4.view.ViewPager;
+
+// Vertical ViewPager implement, original code from
+// https://android.googlesource.com/platform/packages/apps/DeskClock/+/master/src/com/android/deskclock/VerticalViewPager.java
+public class VerticalViewPager extends ViewPager {
+    private boolean mVertical = false;
+    private GestureDetector mGestureDetector;
+
+    public VerticalViewPager(Context context) {
+        this(context, null);
+    }
+
+    public VerticalViewPager(Context context, AttributeSet attrs) {
+        super(context, attrs);
+        setOrientation(false);
```

Comment:
Do we need to call this here? I think to have horizontal ViewPager to behave as it was, we don't really need to do everything that `setOrientation` method is doing, right?

Reply:
I dont want to change original ```ReactViewPager.java``` much more, so it's necessary with some hooks, ```setPageTransformer``` and ```GestureDetector``` shoud be initialized.